src/scheduler/MpiWorld.cpp

```diff
@@ -272,6 +209,22 @@ std::string MpiWorld::getHostForRank(int rank)
     }
 }
 
+// Prepare the host-rank map with a vector containing _all_ ranks
+void MpiWorld::setHostForRank(const std::vector<std::string>& rankHostVec)
+{
+    assert(rankHostVec.size() == size);
+    faabric::util::FullLock lock(worldMutex);
+    for (int i = 0; i < size; i++) {
+        auto it = rankHostMap.try_emplace(i, rankHostVec.at(i));
+        if (!it.second) {
+            logger->error("Error emplacing in rankHostMap: {} -> {}",
+                          i + 1,
+                          rankHostVec.at(i));
+            throw std::runtime_error("Error emplacing in rankHostMap");
```

Comment:
I'm not sure I understand the use of this pattern with `try_emplace` instead of just `emplace`. What situation is it guarding against? It's adding 5 lines of error handling code, and even when it fails the message isn't particularly clear. Much like avoiding variable names in comments, it's not good to include them in error messages, as this will be even more confusing if the variable name changes.

Reply:
`try_emplace` might be an overkill here indeed. It does the same as `emplace` (and has the same return value), but the the element is not [constructed if the key is already there](https://en.cppreference.com/w/cpp/container/map/try_emplace) (again, not important with `string`s and `int`s). In short, `emplace` and `try_emplace` guard against the same behaviour: trying to insert a key that already exists.

The error message was not clear at all.

Reply:
Yes I understand what `try_emplace` does, but what situation is it guarding against _here_? It seems like this will only error if we're setting a value for a rank that has already been set. AFAICT this won't happen as it's only called once in MPI initialisation right? Even if it did somehow happen (e.g. if someone changes the code in future), it's probably not worthy of an error, at most a warning to say that it's getting reset. Either way I think just using `emplace` is fine and much clearer.